### PR TITLE
vehicle: fix sigfpe when acceleration is zero

### LINKED ISSUE #45473: SIGFPE: Arithmetical error in vehicle::thrust
# Describe the bug

I've got SIGFPE when applied handbrake for electrobike with empty batteries and exhausted character. (or push some random keys when running from the horde)

# Steps To Reproduce

Reproduction steps are posted in the first comment.

# Expected behavior

Handle integer division correctly.

# Versions and configuration

 - OS: Ubuntu 18.04
 - Game Version: 0.E-7497-g48b6f14
 - Graphics version: Tiles
 - Ingame language: Russian
 - Mods loaded: bunch of, but it doesn't matter

# Additional context

Backtrace:
```    ./cataclysm-tiles(_Z21debug_write_backtraceRSo+0x38) [0xc015c5]
    ./cataclysm-tiles() [0xbeb402]
    ./cataclysm-tiles() [0xbeb584]
    /lib/x86_64-linux-gnu/libc.so.6(+0x3efd0) [0x7f9778e8afd0]
    ./cataclysm-tiles(_ZN7vehicle6thrustEii+0x30f) [0x13bcaa3]
    ./cataclysm-tiles(_ZN7vehicle10gain_movesEv+0xe5) [0x13a88db]
    ./cataclysm-tiles(_ZN3map7vehmoveEv+0x112) [0xf291a6]
    ./cataclysm-tiles(_ZN4game7do_turnEv+0x86a) [0xceebfc]
    ./cataclysm-tiles(main+0x139c) [0x91f23a]
    /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f9778e6db97]
    ./cataclysm-tiles(_start+0x29) [0x9ef539]

    Attempting to repeat stack trace using debug symbols…
    debug_write_backtrace(std::ostream&)
    ??:?
    log_crash
    crash.cpp:?
    signal_handler
    crash.cpp:?
    ??
    ??:0
    vehicle::thrust(int, int)
    ??:?
    vehicle::gain_moves()
    ??:?
    map::vehmove()
    ??:?
    game::do_turn()
    ??:?
    main
    ??:?
    ??
    ??:0
    _start
    ??:?
```

Disasm:
```
 13bca8f:   sub    $0x7d0,%eax      ; eax -= 20*100
 13bca94:   mov    $0x0,%edx
 13bca99:   cmovs  %edx,%eax        ; eax = eax < 0 ? 0 : eax
 13bca9c:   imul   $0x3e8,%eax,%eax ; eax *= 1000
 13bcaa2:   cltd
 13bcaa3:   idivl  (%rsp)           ; eax /= local_var
```
This is exactly this line of code:
https://github.com/CleverRaven/Cataclysm-DDA/blob/8d87f156b966b53a4dd5ae3e1bcc4dfbccb67685/src/vehicle_move.cpp#L466

I would suggest to add `&& accel != 0` into this condition
https://github.com/CleverRaven/Cataclysm-DDA/blob/8d87f156b966b53a4dd5ae3e1bcc4dfbccb67685/src/vehicle_move.cpp#L454
but I'm not very familiar with the code, just performed quick investigation of what has happened.

### PR DESCRIPTION
**Pull request by CAHEK7**
#### Summary

SUMMARY: Bugfixes "Fix sigfpe crash in vehicle::thrust"

#### Purpose of change

It fixes rare case when all engines produce zero power due to no fuel or exhausted character and target cruise speed is set less than current.
Fix #45473

#### Describe the solution

Just jump over the case where division by 0 occurs, set `load = 0` inside `else` branch and perform all other calculations.

#### Describe alternatives you've considered

Maybe it should be done a few lines earlier to perform early exit: remove `thrusting` from `if( thrusting && accel == 0 )`.

#### Testing

Just checked that save from #45473 doesn't crash anymore and ran `make check`

#### Additional context

I just fixed exact place where the crash occurs.
The fix is kind of trivial but yet can be done in many other ways like move this check into some other place or even special handling zero-accelerated vehicles and I'm not very familiar with the codebase so I can miss something important.

### CODE REVIEW

**Comment by kevingranade** [comment]
Thanks for finding the source, this unfortunately a pretty tricky chump of logic, so we need to be careful.

Just summarizing some stuff:
accel == 0 at that line implies thrusting == false
This means that the "sgn" of acceleration and velocity are different, meaning we're braking instead of accelerating.
accel == 0 means vehicle::acceleration() returned 0, the most obvious cause of which is that the vehicle is unfueled and/or the engines are off.
The only outcome of calculating load is to put strain on the engine, but in this scenario the engine is braking, so that won't happen anyway.

So tl;dr, looks like your change does the right thing.
